cranelift/codegen/src/machinst/lower.rs

```diff
@@ -775,6 +777,14 @@ impl<'func, I: VCodeInst> Lower<'func, I> {
             let loc = self.srcloc(inst);
             self.finish_ir_inst(loc);
 
+            // TODO maybe insert random instruction
+            if ctrl_plane.get_decision() {
+                let imm: u64 = ctrl_plane.get_arbitrary();
+                let regs = self.alloc_tmp(crate::ir::types::I64);
```

Comment:
We could perhaps choose a float or int here, so we get the increased register pressure / potential clobbering of both kinds of registers. I guess that implies two kinds of `gen_imm` as well.